contracts/src/challenge/SymChallenge.sol

```diff
@@ -86,23 +86,46 @@ contract SymChallenge is ChallengeBase, ISymChallenge {
         daProvider = _daProvider;
         resultReceiver = _resultReceiver;
         startStateHash = _startStateHash;
-        endStateHash = _endStateHash;
 
         turn = Turn.Defender;
         lastMoveBlock = block.number;
         defenderTimeLeft = challengePeriod;
         challengerTimeLeft = challengePeriod;
     }
 
-    // TODO: use minimum of challenger/defender's proposed numSteps.
-    function initializeChallengeLength(uint256 _numSteps) external override onlyOnTurn {
+    function proposeChallenge(bytes32 _endStateHash, uint256 _numSteps) external {
```

Comment:
what happens in the special case where they both pass in the same hashes but for different `numSteps`:
```
// defender calls:
proposeChallenge(sameStateHash, 32)
// challenger calls:
proposeChallenge(sameStateHash, 64)
```